src/pruner_util.cpp

```diff
@@ -75,20 +75,70 @@ cstring remove_extension(cstring file_path) {
 
 int get_exit_code(cstring name, P4PRUNER::PrunerConfig pruner_conf) {
```

Comment:
Let's untangle this a bit. I think we should handle this in two phases.
First, we fill the information for the config. This the exit code and the error message. Then we use this information to check stability every time. In the case of validation, the error message is just an empty string for now.
Let's change `get_exit_code` to `get_exit_info` and it will return a struct or tuple containing this exact information. We call it at the beginning to fill the config information and then reuse it in subsequent passes.